components/clp-py-utils/clp_py_utils/clp_config.py

```diff
@@ -335,13 +420,23 @@ def validate_target_segment_size(cls, field):
             raise ValueError("target_segment_size must be greater than 0")
         return field
 
-    def make_config_paths_absolute(self, clp_home: pathlib.Path):
-        self.directory = make_config_path_absolute(clp_home, self.directory)
+    def set_directory(self, directory: pathlib.Path) -> None:
+        storage_config = self.storage
+        if StorageType.FS == storage_config.type:
+            storage_config.directory = directory
+        else:
+            storage_config.staging_directory = directory
```

Comment:
To detect programmer errors in the future, we should do `elif StorageType.S3 == ...` and add an `else` which raises `NotImplementedError`. Ditto in `get_directory`.